pyccel/ast/variable.py

```diff
@@ -219,49 +219,72 @@ def __init__(
             self._order = None
 
     def process_shape(self, shape):
-        """ Simplify the provided shape and ensure it
-        has the expected format
+        """
+        Simplify the provided shape and ensure it has the expected format.
+
+        The provided shape is the shape used to create the object. In most
+        cases where the shape is required we do not require this expression
+        (which can be quite long). This function therefore replaces those
+        expressions with calls to PyccelArrayShapeElement.
```

Comment:
I think it is worth mentioning that the original expression is not only removed in case it is long but also because it can become invalid. Eg:
```python
n = 2
x = np.ones(n) # _shape = Variable('n') ok here
n = 3
print(x.shape[0]) # _shape = Variable('n') wrong here
```

Reply:
Fixed in fcfd775